### PR TITLE
Feature/refresh token

### PR DESCRIPTION
**Pull request by wesleymonte**

### CODE REVIEW

File: src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java [hunk 1]
```diff
@@ -10,6 +11,9 @@
 
 public class OAuthToken {
 
+    @SerializedName("version")
```
Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java]: Replace by constant.

File: src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java [hunk 2]
```diff
@@ -29,6 +33,16 @@
     public OAuthToken() { }
 
     public OAuthToken(String accessToken, String refreshToken, String userId, Date expirationDate) {
+        this.version = 0L;
```
Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java]: Replace by constant.

File: src/main/java/org/fogbowcloud/app/jdfcompiler/job/JDFJobBuilder.java [hunk 3]
```diff
@@ -213,36 +213,45 @@ private void parseFinalCommands(String jobId, TaskSpecification taskSpec, Task t
...
+				+ " if [ $http_code == " + HttpStatus.UNAUTHORIZED.toString() + " ] ; then "
+			  + "token=$(" + requestTokenCommand + "); "
+				+ downloadCommand + " fi";
 
 		return new Command(scpCommand, rawCommand);
 	}
 
+	private String getRefreshTokenCommand(){
+		final String iguassuUrl = this.properties.getProperty(IguassuPropertiesConstants.IGUASSU_SERVER_HOST);
+		final String refreshTokenUrl = String.format("%s/auth/refresh/${token}", iguassuUrl);
```
Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/jdfcompiler/job/JDFJobBuilder.java]: Pass the token in a header like `X-Expired-Token`.

Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/jdfcompiler/job/JDFJobBuilder.java]: There may be a missing part in the string of this endpoint. Would not it be `/ auth / oauth2 / refresh`?

File: src/main/resources/templates/iguassu.conf.template [hunk 4]
```diff
@@ -15,6 +15,7 @@ oauth_storage_service_client_id=
 oauth_storage_service_client_secret=
 
 #### Storage Service ####
+#Requires adding http
```
Comment (emanueljoivo) [inline src/main/resources/templates/iguassu.conf.template]: Replace by : 'Requires http prefix `http://`'

File: src/main/java/org/fogbowcloud/app/core/datastore/OAuthTokenDataStore.java [hunk 5]
```diff
@@ -207,11 +211,13 @@ public OAuthToken getObjFromDataStoreResult(ResultSet rs) throws SQLException {
         String ownerUsername = rs.getString(USER_ID);
```
Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/core/datastore/OAuthTokenDataStore.java]: These properties can be final.

File: src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java [hunk 6]
```diff
@@ -120,4 +119,58 @@ private String generateIguassuToken(String userId) {
...
+            final String iguassuToken = this.generateIguassuToken(refreshedToken.getUserId());
+            this.storeOAuthToken(refreshedToken, iguassuToken);
+
+            final long oldVersions = oAuthToken.getVersion() - 1;
+            if(oldVersions >= 0){
+                this.iguassuFacade.removeOAuthTokens(oAuthToken.getUserId(), oldVersions);
+            }
+            return refreshedToken.getAccessToken();
+        } else {
+            throw new NotFoundAccessToken("The accessToken[" + accessToken +"] was not found");
```
Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java]: Treat this as an unauthorized operation.

File: src/main/java/org/fogbowcloud/app/api/constants/ApiDocumentation.java [hunk 7]
```diff
@@ -8,6 +8,7 @@
         public static final String JOB_PATH = "/{jobId}";
         public static final String TASKS_ENDPOINT = "tasks";
         public static final String OAUTH2_ENDPOINT = "/oauth2";
```
Comment (emanueljoivo) [inline src/main/java/org/fogbowcloud/app/api/constants/ApiDocumentation.java]: Attempt for this.

File: src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java (hunk 1 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java]: Done.

File: src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java (hunk 2 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/app/core/datastore/OAuthToken.java]: Done.

File: src/main/java/org/fogbowcloud/app/jdfcompiler/job/JDFJobBuilder.java (hunk 3 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/app/jdfcompiler/job/JDFJobBuilder.java]: Token no longer required for request.

Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/app/jdfcompiler/job/JDFJobBuilder.java]: yes. added.

File: src/main/java/org/fogbowcloud/app/core/datastore/OAuthTokenDataStore.java (hunk 5 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/app/core/datastore/OAuthTokenDataStore.java]: Do not need.

File: src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java (hunk 6 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/app/api/http/services/AuthService.java]: Okay. Done.

**Pull request merged**